datafusion-cli/src/print_format.rs

```diff
@@ -209,6 +211,145 @@ impl PrintFormat {
         }
         Ok(())
     }
+
+    #[allow(clippy::too_many_arguments)]
```

Comment:
most of the newly added functions don't use fields on  `PrintFormat`

I think we could probably avoid the 
```rust
    #[allow(clippy::too_many_arguments)]
```

By encapsulating all the `mut` variables here in a new struct

Maybe something like

```rust
struct OutputStreamState {
        preview_batches: Vec<RecordBatch>,
        preview_row_count: usize,
        preview_limit: usize,
        precomputed_widths: Option<Vec<usize>>,
        header_printed: bool,
        writer: &mut dyn std::io::Write,
}

impl OutputStreamState {
  fn process_batch(&mut self, batch: RecordBatch) { ... }
...
  fn compute_column_widths(&self, 
    batches: &Vec<RecordBatch>,
    schema: SchemaRef,
  ) {...}

}

```

Reply:
Thanks @alamb for review, this is a very good suggestion!

Reply:
Created a follow-up ticket:
https://github.com/apache/datafusion/issues/14886